require.js

```diff
@@ -69,13 +67,11 @@
     if (loaderPath == null) loaderPath = '/';
     path = unalias(name, loaderPath);
 
+    if (!has.call(modules, path)) path = aliases[path];
+
     if (has.call(cache, path)) return cache[path].exports;
     if (has.call(modules, path)) return initModule(path, modules[path]);
 
-    var dirIndex = expand(path, './index');
-    if (has.call(cache, dirIndex)) return cache[dirIndex].exports;
-    if (has.call(modules, dirIndex)) return initModule(dirIndex, modules[dirIndex]);
```

Comment:
Because we're creating aliases for registered files we don't need to do an extra check for index. The alias will find it.